src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -211,16 +203,27 @@ private ITestAgent CreateRemoteAgent(TestPackage package, int waitTime)
             return null;
         }
 
-        private static string GetTestAgentExePath(bool requires32Bit)
+        private static string GetTestAgentExePath(RuntimeFramework targetRuntime, bool requires32Bit)
         {
             string engineDir = NUnitConfiguration.EngineDirectory;
             if (engineDir == null) return null;
 
+            string agentsDir = Path.Combine(engineDir, "agents");
+            log.Debug($"Checking for agents at {agentsDir}");
+
+            if (!Directory.Exists(agentsDir))
+            {
+                agentsDir = Path.Combine(Path.GetDirectoryName(engineDir), "agents");
+                log.Debug($"Directory not found! Using {agentsDir}");
```

Comment:
Minor: How about removing the second calculation of agentsDir? Just thinking this may break in future, if someone alters one line, but not the other. 🙂 (I also think this will actually log the wrong path - `engineDir` is already a directory, so calling `Path.GetDirectoryName` on it will get the directory a level up)


e.g.
```suggestion
                log.Debug($"Directory not found! Using {agentsDir}");
```

Reply:
What you say about getting the parent dir is exactly what's intended! We run in two environments. When running tests in the output directory, we need a different relative path from what we do in the packages. I'll verify that this is correct first and then add a comment since it is a bit confusing.

Reply:
Confirmed and added a comment. The two calculations give two diferent results, as intended.

Reply:
Thanks, that makes more sense. Thanks for adding the comment